tests/suites/test_suite_net.function

```diff
@@ -26,3 +70,63 @@ void context_init_free( int reinit )
     goto exit;
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PLATFORM_IS_UNIXLIKE */
+void poll_beyond_fd_setsize( )
+{
+    /* Test that mbedtls_net_poll does not misbehave when given a file
+     * descriptor beyond FD_SETSIZE. This code is specific to platforms
+     * with a Unix-like select() function. */
+
+    struct rlimit rlim_nofile;
+    int restore_rlim_nofile = 0;
+    int ret;
+    mbedtls_net_context ctx;
+    uint8_t buf[1];
+
+    mbedtls_net_init( &ctx );
+
+    /* On many systems, by default, the maximum permitted file descriptor
+     * number is less or equal to FD_SETSIZE. If so, raise the limit if
```

Comment:
IIUC, if the maximum permitted fd number is equal to `FD_SETSIZE` then things are fine. OTOH, `RLIMIT_NOFILE` is one more than the max permitted fd number, so we want it to be at least `FD_SETSIZE + 1`.